File: Data Structures & Algorithms/word-pattern/submission-1.py
class Solution:
    def wordPattern(self, pattern: str, s: str) -> bool:
        p_map = {}
        s_map = {}
        s = s.split()

        if len(pattern) != len(s):
            return False

        
        for i in range(min(len(pattern), len(s))):
            if (s[i] in s_map and s_map[s[i]] != pattern[i]) or (pattern[i] in p_map and p_map[pattern[i]] != s[i]):
                return False
            p_map[pattern[i]] = s[i]
            s_map[s[i]] = pattern[i]

        return True

File: Data Structures & Algorithms/is-anagram/submission-1.go
func isAnagram(s string, t string) bool {
	if len(s) != len(t) {
		return false
	}

	sMap := make(map[string]int)
	tMap := make(map[string]int)

	for i, _ := range s {
		sMap[string(s[i])] = sMap[string(s[i])] + 1
		tMap[string(t[i])] = tMap[string(t[i])] + 1
	}

	for k, v := range sMap {
		if tMap[k] != v {
			return false
		} 
	}

	return true
}
